src/types/vec/vec.zig

```diff
@@ -698,16 +698,16 @@ fn SignedNumericVec3Mixin(comptime Self: type, comptime Dimens: comptime_int, co
             /// Modifies vector itself so that it is a cross product of vectors, i.e. a vector that is perpendicular to the plane
             /// containing x and y and has a magnitude that is equal to the area of the parallelogram that x and y span.
             pub fn crossSelf(a: *Self, b: Self) void {
-                a.x = a.y * b.z - a.z * b.y;
```

Comment:
I would say that the general "rule" is that "*Self" methods have most of the logic, so I would recommend making a const copy of the first parameter in crossSelf as you did in the first commit.